model/DataObject.php

```diff
@@ -1790,17 +1826,12 @@ public function has_one($component = null) {
 	 *        the field data stripped off. It defaults to TRUE.
 	 * @return string|array
 	 */
-	public function belongs_to($component = null, $classOnly = true) {
-		$belongsTo = $this->config()->belongs_to;
-
+	public function belongsTo($component = null, $classOnly = true) {
 		if($component) {
-			if($belongsTo && array_key_exists($component, $belongsTo)) {
-				$belongsTo = $belongsTo[$component];
-			} else {
-				return false;
-			}
+			return $this->belongsToComponent($component, $classOnly);
 		}
 
+		$belongsTo = $this->stat('belongs_to');
```

Comment:
Are we meant to be using `stat`? I know it uses `Config` but isn't that just for legacy? Other places in this refactor are using uninherited config settings, which seems more appropriate?